lib/components/form/call-taker/date-time-picker.tsx

```diff
@@ -134,7 +137,20 @@ const DateTimeOptions = ({
   // Update state when external state is updated
   useEffect(() => {
     if (initialDate !== date) setDate(initialDate)
-    if (initialTime !== time) setTime(initialTime)
+    if (initialTime !== time) {
+      setTime(initialTime)
+
+      // Don't update if still typing
+      if (timeRef.current !== document.activeElement) {
```

Comment:
Seems to work now, but would it be simpler to handle `onBlur` instead?

Reply:
I tried this, but some edge cases are missed when doing this